mod/file/views.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+return call_user_func(function() {
+	$dir = dir(__DIR__ . '/graphics/icons');
+	$ret = [];
+
+	while (false !== ($entry = $dir->read())) {
+		if ($entry[0] === '.') {
+			continue;
+		}
+
+		$ret['default']["file/icons/$entry"] = "{$dir->path}/$entry";
+	}
+
+	return $ret;
+});
```

Comment:
I do still prefer core the way it is if we can keep it that way...

Actually I just had a thought, what if we used .htaccess in plugins to forward through simplecache? Wouldn't work for nginx but could be a fix for the theming problem I described above for apache, which seems like the most common configuration.

Might be a bad idea to have such different behavior depending on web server though...